### ISSUE TITLE
COCO2017 training report

### ISSUE THREAD

**Issue by thoang3** [NVIDIA/retinanet-examples#74]
Hi everyone, 

I am able to train COCO2017 based on the instructions. However, this issue occurs at around 8000 iterations. I have checked and the image is there. Does anyone know the possible reason for this? Perhaps the image is corrupted somehow?
Thank you in advance!

<img width="1273" alt="RTNN training issue" src="https://user-images.githubusercontent.com/9111479/66679281-b61e6800-ec22-11e9-9720-e7f8f92b8b92.png">

**Comment by james-nvidia** [comment]
The `FileNotFoundError` suggests that the file is either not there, or can't be opened.
Can you open the image and look at it?
It looks like you are passing an extra `/` to the validation images folder. Please try without the tailing `/`.

**Comment by thoang3** [comment]
Thank you for your quick response! The image is there, and can be opened and viewed (attached below). I've also tried to verify the image via PIL with img.verify() and there's no issue. 

Validation data was downloaded via this [link](http://images.cocodataset.org/zips/val2017.zip).
Annotation was downloaded via this [link](http://images.cocodataset.org/annotations/annotations_trainval2017.zip).

So the issue might originate from the extra / that you mentioned. However, I haven't changed anything but used your provided code, which is quite strange. Let me look at the code to see if I could find anything unusual. 

![000000312213](https://user-images.githubusercontent.com/9111479/66681236-24652980-ec27-11e9-947d-1ea0c398ff2e.jpg)

**Comment by james-nvidia** [comment]
What happens if you try and open the image using PIL?
```
from PIL import Image
filepath = '/home/???????/coco/images/val2017/000000312213.jpg'
im = Image.open(filepath).convert("RGB")
```

**Comment by thoang3** [comment]
Hi James! 

Thank you for your comment! I've just found out the other day that it's actually my mistake when symlink the validation dataset, and there's nothing wrong with the image itself.

Btw, I'd like to report my training results on COCO as follow:

1. Training COCO2017 by default parameters (resnet50, 90000 iterations, ...)
2. When training with 1 GPU (RTX 20180 Ti), the loss always diverge no matter how low I set up the learning rate (down to 0.000001), so it's similar to other reports, and it seems like we need to come up with some warm-up strategy for training with 1 GPU. Another extra thing is that, when I ran into the silly mistake above, I was training the default model with 2 GPUs, and it was converging as well (until 8000 iterations then it got stuck since it couldn't find validation data). So perhaps the current config might work even for 2 GPUs. 
3. When training with 6 GPUs (RTX 2080 Ti) with default parameters, the loss is quite healthy as illustrated in attached figures below. 

<img width="583" alt="Before jump Screen Shot 2019-10-14 at 10 00 23 AM" src="https://user-images.githubusercontent.com/9111479/66770349-41d70500-ee6c-11e9-82aa-1d1f7f420b9d.png">
<img width="584" alt="Jump Screen Shot 2019-10-14 at 10 00 42 AM" src="https://user-images.githubusercontent.com/9111479/66770353-46032280-ee6c-11e9-9221-59d655c24d99.png">
<img width="580" alt="90000 Screen Shot 2019-10-14 at 9 57 59 AM" src="https://user-images.githubusercontent.com/9111479/66770360-48fe1300-ee6c-11e9-8233-0872ffb4f695.png">

4. As you can see, after 90000 iterations, the final mAP is 33%, which is a little lower than the posted number which is 35%. I figure this might be due to the current config being optimized for 8 GPUs. Since the loss still converges steadily, I am currently running for 30000 iterations more (up to 120000 iterations). I will share the result soon.
5. Another fun thing I notice is the big jump in mAP (from 27% to 31.8% for mAP, and from 43.5% to 49.8% for mAP@50) only after one round of validation. Perhaps the loss has just "escaped" from a local minimum and found a much better one :).

**Comment by james-nvidia** [comment]
I'm glad you solved the image problem. I'll close this issue now.
We don't have much experience training on 1 GPU. Please let us know which parameters you find effective, and your final mAP for your current training run.
Thanks!

**Comment by thoang3** [comment]
Hi everyone!

I'd like to keep you posted that mAP can reach 34% ish (trained up to 240000 iterations), which is great! 

<img width="502" alt="240000" src="https://user-images.githubusercontent.com/9111479/66970665-1a866080-f043-11e9-8526-13d26c5d830d.png">
